Docs move keyless info to overview (#156)
sigstore/docs: content/en/cosign/overview.md

To verify, Cosign queries the transparency log (Rekor) to compare the public key bound to the certificate, and checks the timestamp on the signature against the artifact’s entry in the transparency log. The signature is valid if its timestamp falls within the small window of time that the key pair and certificate issued by the certificate authority were valid.

## Working with containers

> > NOTE: You will need access to a container registry for cosign to work with. [ttl.sh](https://ttl.sh/) offers free, short-lived (i.e., hours), anonymous container image hosting if you just want to try these commands out.

Reviewer: I don't think we need this content in the overview, lines 68-77. I would suggest removing it entirely, I don't think it adds anything. If someone does think there's value, then we could put it in signing_with_containers.md.

Author: rewrote to make this an example.

Using `ttl.sh` and [crane](https://github.com/google/go-containerregistry/tree/main/cmd/crane) to prepare the image that we want to sign. Run the following:

```
$ SRC_IMAGE=busybox
$ SRC_DIGEST=$(crane digest busybox)
$ IMAGE_URI=ttl.sh/$(uuidgen | head -c 8 | tr 'A-Z' 'a-z')
$ crane cp $SRC_IMAGE@$SRC_DIGEST $IMAGE_URI:1h
$ IMAGE_URI_DIGEST=$IMAGE_URI@$SRC_DIGEST
```

### Keyless signing of container

```

> $ cosign sign $IMAGE_URI_DIGEST

Reviewer: Given this is an overview section, I would revert back to only including `cosign sign <image URI>` without the output. This is already in signing_with_blobs.md and signing_with_containers.md.

Author: took out the output.

	This may include the email address associated with the account with which you authenticate.
	This information will be used for signing this artifact and will be stored in public transparency logs and cannot be removed later.

By typing 'y', you attest that you grant (or have permission to grant) and agree to have this information stored permanently in transparency logs.
Are you sure you would like to continue? [y/N] y
tlog entry created with index: 18914440
Pushing signature to: ttl.sh/4d6d55ae
```

### Keyless verifying of container

This works the same as verifying a blob, but the signature and certificate are attached as container metadata, so there is no need to place the certificate and signature on the verify command. To verify a signed container image, use the following command:

```
$ cosign verify <image URI> --certificate-identity=name@example.com
                            --certificate-oidc-issuer=https://accounts.example.com
```

> Note that in the following example we use the `regexp` versions of the identity options:

```
cosign verify $IMAGE_URI_DIGEST --certificate-identity-regexp=.* --certificate-oidc-issuer-regexp=.*

> Verification for ttl.sh/4d6d55ae@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16 --

Reviewer: Same comment here, I think we should revert back to a generic example.

Author: keeping the ttl.sh example.

To verify, Cosign queries the transparency log (Rekor) to compare the public key bound to the certificate, and checks the timestamp on the signature against the artifact’s entry in the transparency log. The signature is valid if its timestamp falls within the small window of time that the key pair and certificate issued by the certificate authority were valid.

## Example: Working with containers

> NOTE: In this example, we will create a container using [ttl.sh](https://ttl.sh/).  It offers free, short-lived (i.e., hours), anonymous container image hosting if you just want to try out signing and verifying commands.  

To use `ttl.sh` and [crane](https://github.com/google/go-containerregistry/tree/main/cmd/crane) to prepare the image that we want to sign. Run the following:

```
$ SRC_IMAGE=busybox
$ SRC_DIGEST=$(crane digest busybox)
$ IMAGE_URI=ttl.sh/$(uuidgen | head -c 8 | tr 'A-Z' 'a-z')
$ crane cp $SRC_IMAGE@$SRC_DIGEST $IMAGE_URI:1h
$ IMAGE_URI_DIGEST=$IMAGE_URI@$SRC_DIGEST
```

### Keyless signing of a container

> The following code signs the created container image.  The command to sign container images is `cosign sign <image URI>`.  Note that for containers, there is no bundle as there is with blobs, as the signature and certificate are attached directly to the container:

Reviewer 2: I would be consistent with the variable

```suggestion
The following code signs the created container image.  The command to sign container images is `cosign sign $IMAGE_URI_DIGEST`.  Note that for containers, there is no bundle as there is with blobs, as the signature and certificate are attached directly to the container:
```

Author: That's not exactly the command to sign images.   It's actually cosign sign <image URI> not $IMAGE_URI_DIGEST.   That is just the command for this example.  I can change the wording so it says it's the command for the example.

Author: I reworded to indicate that the example uses $IMAGE_URI_DIGEST.

```
$ cosign sign $IMAGE_URI_DIGEST
```

### Keyless verifying of a container

> This works the similar to verifying a blob, but there is no need to place the certificate and signature on the verify command. To verify a signed container image, use the following command:

Reviewer 2: ```suggestion
This works similarly to verifying a blob, but there is no need to place the certificate and signature on the `cosign verify` command. To verify a signed container image, use the following command:
```

```

> $ cosign verify <image URI> --certificate-identity=name@example.com

Reviewer 2: Consistency here?

```suggestion
$ cosign verify $IMAGE_URI_DIGEST --certificate-identity=name@example.com
```

Author: Changing it like this would imply to readers that they should type $IMAGE_URI_DIGEST in the command, and that's not correct.

tlog entry created with index: 18914440
Pushing signature to: ttl.sh/4d6d55ae
```

### Keyless verifying of a container

This works the similar to verifying a blob, but there is no need to place the certificate and signature on the verify command. To verify a signed container image, use the following command:

```
$ cosign verify <image URI> --certificate-identity=name@example.com
                            --certificate-oidc-issuer=https://accounts.example.com
```

> Note that for our example we use the `regexp` versions of the identity options:

```
cosign verify $IMAGE_URI_DIGEST --certificate-identity-regexp=.* --certificate-oidc-issuer-regexp=.*

Verification for ttl.sh/4d6d55ae@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates

[{"critical":{"identity":{"docker-reference":"ttl.sh/4d6d55ae"},"image":{"docker-manifest-digest":"sha256:b5d6f2a0dc1"

...
```

> The rest of the flags (annotations, claims, tlog, etc.) all work the same.

Reviewer: Can we delete lines 125-132, these three paragraphs? All of this information is already discussed in the intro to the quickstart.

Author: removed

Enter password for private key:
Pushing signature to: index.docker.io/user/demo:sha256-87ef60f558bad79be4def8.sig
```
## SCM Integration

> Cosign integrates natively with source code management (SCM) systems like GitHub and GitLab. You can use the official [GitHub Actions Cosign installer](https://github.com/marketplace/actions/cosign-installer) or use cosign to generate and work safely with [SCM secrets](/cosign/git_support/) with native API integration.

Reviewer 2: ```suggestion
Cosign integrates natively with source code management (SCM) systems like GitHub and GitLab. You can use the official [GitHub Actions Cosign installer](https://github.com/marketplace/actions/cosign-installer) or use Cosign to generate and work safely with [SCM secrets](/cosign/git_support/) with native API integration.
```

$ cosign sign --identity-token=$IDENTITY_TOKEN $IMAGE_DIGEST
```

From outside a GCE VM, you can impersonate a GCP IAM service account to sign an image:

```
$ IDENTITY_TOKEN=$(gcloud auth print-identity-token \
        --audiences=sigstore \
        --include-email \
        --impersonate-service-account my-sa@my-project.iam.gserviceaccount.com)
$ cosign sign --identity-token=$IDENTITY_TOKEN $IMAGE_DIGEST
```

In order to impersonate an IAM service account, your account must have the
`roles/iam.serviceAccountTokenCreator` role.

> Note: On Google Cloud Build, standard identity tokens are not supported through the GCE metadata server.
Cosign has a special flow for this case, where you can instruct the Cloud Build service account to impersonate
another service account.
To configure this flow:

1. Create a service account to use for signatures (the email address will be present in the certificate subject).
2. Grant the Cloud Build service account the `roles/iam.serviceAccountTokenCreator` role for this target account.
3. Set the `GOOGLE_SERVICE_ACCOUNT_NAME` environment variable to the name of the target account in your cloudbuild.yaml
4. Sign images in GCB, without keys!

> ### Timestamps

Reviewer: Can we create a new file called "timestamping.md" and move this content to that? We need to flesh that out a lot more, but I don't think this makes sense in the overview.

Signature timestamps are checked in the [rekor](https://github.com/sigstore/rekor) transparency log. Rekor's `IntegratedTime` is signed as part of its `signedEntryTimestamp`. Cosign verifies the signature over the timestamp and checks that the signature was created while the certificate was valid.

### Other Formats

Cosign is useful not only for blobs, containers, and container-related artifacts; it can also be used for other file types.

To learn how to sign SBOMs, WASM modules, Tekton bundles and more, review [Signing Other Types](/cosign/other_types/). For more information about blobs, review [Working with Blobs](/cosign/working_with_blobs/).

> ## Custom Components

Reviewer: I think we should "custom components" under https://github.com/sigstore/docs/blob/main/content/en/cosign/verify.md

Author: moved

Pushing signature to: index.docker.io/user/demo:sha256-87ef60f558bad79be4def8.sig
```

### Other Formats
## Other Formats

Cosign is useful not only for blobs, containers, and container-related artifacts; it can also be used for other file types.

To learn how to sign SBOMs, WASM modules, Tekton bundles and more, review [Signing Other Types](/cosign/other_types/). For more information about blobs, review [Working with Blobs](/cosign/working_with_blobs/).

> ### SCM Integration

Reviewer: Can we leave these two sections in this quickstart for now? I think we can figure out a better place for them later, but I don't think they should be in verify.md.

Let's also update header to ## rather than ###

Cosign integrates natively with source code management (SCM) systems like GitHub and GitLab. You can use the official [GitHub Actions Cosign installer](https://github.com/marketplace/actions/cosign-installer) or use cosign to generate and work safely with [SCM secrets](/cosign/git_support/) with native API integration.

### Attestations

In addition to signatures, Cosign can be used with [In-Toto Attestations](https://github.com/in-toto/attestation).
To learn how to sign SBOMs, WASM modules, Tekton bundles and more, review [Signing Other Types](/cosign/other_types/). For more information about blobs, review [Signing Blobs](/cosign/signing_with_blobs/). For containers, see [Signing Containers](/cosign/signing_with_containers/).

> Attestations provide an additional semantic-layer on top of plain cryptographic signatures that can be used in policy systems.

Reviewer: Can you add the line: 

Learn more in the [Attestations](https://docs.sigstore.dev/cosign/attestation) documentation.

Author: added
